Use three way diff in git merges to make conflict resolution easier

Stratus3D opened the pull request:
The diff3 setting causes git generate three way diffs when conflicts occur. Normally git diffs just show HEAD and conflicting commit. Three way diffs show HEAD, the conflicting commit, and the parent commit. This is very useful when rebasing.

joshukraine commented:
Cool, thanks! So is that similar to the functionality provided by [vim-conflicted](https://github.com/christoomey/vim-conflicted)?

Stratus3D commented:
Yes, it's like vim-conflicted's three way view of a file containing conflicts, but it's in the diff git generates in the file. 

Three way diff:
```
<<<<<<< HEAD
original line
added line
||||||| merged common ancestors
original line
=======
>>>>>>> topic
```

Default diff:
```
<<<<<<< HEAD
original line
added line
=======
>>>>>>> topic
```

The three way diff makes it clear that the reason a conflict occurred is because a line was added in `HEAD` and in the other commit that section was removed.
